include/trieste/rewrite.h

```diff
@@ -1119,19 +1119,17 @@ namespace trieste
 
   inline Node operator<<(Node node, detail::RangeContents range_contents)
   {
-    for (auto it = range_contents.range.first;
-         it != range_contents.range.second;
-         ++it)
+    for (Node n : range_contents.range)
```

Comment:
To avoid unnecessary RC manipulations can you use a reference?
```suggestion
    for (Node& n : range_contents.range)
```